config.go

```diff
@@ -809,14 +1051,33 @@ func LoadConfig(interceptor signal.Interceptor) (*Config, error) {
 
 	// Finally, parse the remaining command line options again to ensure
 	// they take precedence.
-	flagParser := flags.NewParser(&cfg, flags.Default)
+	flagParser := flags.NewParser(&preCfg, flags.Default)
 	if _, err := flagParser.Parse(); err != nil {
 		return nil, err
 	}
 
+	// Merge the loaded config into the main Config type.
+	cfg, err := mergeConfig(&preCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	// The flag parser above is only aware of the flags in the preCfg
+	// definition, and not necessarily those in the Config struct. To handle
+	// this, we create a new flag parser for the Config struct, as it is
+	// passed to the ValidateConfig function.
+	// Since some flags in preCfg may not exist in Config, we use
+	// flags.IgnoreUnknown here to avoid errors for those flags. However,
+	// unknown flags are not allowed for the preCfg itself, as the earlier
+	// flag parser will error if such flags are present.
+	mergedFlagParser := flags.NewParser(cfg, flags.IgnoreUnknown)
+	if _, err := mergedFlagParser.Parse(); err != nil {
+		return nil, err
+	}
```

Comment:
Let me know if you can think of a cleaner way of doing this, as this is arguably quite dirty.